### ISSUE TITLE
UnwrapValidatedValue missing dependency for an unused import

### ISSUE THREAD

**Issue by mihalyr** [guardrail-dev/guardrail#1439]
Hi, I've just found Guardrail and trying to use it on a project to build a dropwizard java API. I really like that it keeps things simple and clean. 

One of the problems I faced early on was that I had to figure out on my own what dependencies I need and which versions. Other people already noted this in https://github.com/guardrail-dev/guardrail/issues/865 and based on the comment in that issue I've checked the examples for a dropwizard maven build, but my generated code still does not compile.

The example code was missing the Jersey dependencies, I've added the Dropwizard dependencies for now to fix that. I'm planning to add only the minimum jersey dependencies later so I can make an API package out of the generated code only.

But I still couldn't compile the generated code. The last error I face is the following:

```java
package org.hibernate.validator.valuehandling does not exist
```

The above error is thrown for both the generated Resource and the Handler too from the following import:

```java
import org.hibernate.validator.valuehandling.UnwrapValidatedValue;
```

This import is never used in the generated code (similarly to some other imports like `FormDataParam` in my case). 

What package and version is needed to fix the above compilation error? I've already tried `org.hibernate:hibernate-validator:7.0.4.Final`, `org.hibernate.validator:hibernate-validator:7.0.4.Final` and `io.dropwizard:dropwizard-validation:2.0.28`, but all of them failed.

Thank you in advance.

**Comment by mihalyr** [comment]
Oh, I think I found the problem here https://hibernate.org/validator/documentation/migration-guide/

In `hibernate-validator` `6.0.0.Final` :

> The experimental annotation `org.hibernate.validator.valuehandling.UnwrapValidatedValue`, enum `org.hibernate.validator.valuehandling.UnwrapMode` and the corresponding method `unwrapValidatedValue()` in the constraint declaration API have been removed in the favor of the new `javax.validation.valueextraction.Unwrapping` constraint payloads ([HV-1207](https://hibernate.atlassian.net/browse/HV-1207)).

It seems that Guardrail generates code that depends on an outdated `hibernate-validator` version (`6.0.0.Final` came out on 2017-08-07 and the generated code depends on something even older). The latest version is `7.0.4.Final` from  2022-03-10.

My workaround is to use [5.4.3.Final](https://mvnrepository.com/artifact/org.hibernate/hibernate-validator/5.4.3.Final) from 2019, but it has a list of CVEs from its dependencies listed in the Maven repository. 

Would it be possible to remove or upgrade to a newer hibernate validator version?

**Comment by blast-hardcheese** [comment]
@mihalyr I've got two sort of big projects going currently, #1409 and #1414, though I'm happy to lend a hand if you've got some time to drive this change 🙂

If you're not familiar with JavaParser, the easiest way forward would be to

1. take the non-working output in your sample project,
2. move the source files out of `target/generated-sources` into `src/main`,
3.  then tweak those sources until it both compiles and functions to your liking.

Once you've done that, the resulting diff should be fairly straightforward to replicate in the JavaParser generator, and we can cut a new version based on the most recently released tag to keep you from being blocked by the aforementioned PRs.

(The reason the build is failing on master currently is because I broke backwards compatibility with the previous minor version, so the build will continue to fail until the next minor release is cut, so we'll do a feature branch off `tag: java-dropwizard-v0.67.1`)

**Comment by mihalyr** [comment]
@blast-hardcheese I found my problem was that I used a recent Dropwizard version (`2.0.28`), but Guardrail depends on an older series (`1.3.29`) which uses the older `hibernate-validator` version.

I think this might be a larger upgrade to all these dependencies and Scala is not my thing, I don't think it is a good use of my time, sorry about it. I'll skip Guardrail for server code generation for now. So far the client looked OK.

Regarding this issue, this does not seem to be a bug anymore as the code works fine if I use the older dependency versions. My mistake was to use the new version. So the bug is maybe in the documentation that it doesn't make it clear how to use the tool properly, however that is covered already by https://github.com/guardrail-dev/guardrail/issues/865 I think. Feel free to close this issue.
